python/ic_azure/azure_client.py

```diff
@@ -119,21 +119,32 @@ def resource_client(self):
 
         return self._resource_client
 
-    def kusto_query(self, query):
-        """Run Kusto-query to Azure REST APIs."""
+    def query(self, method="get", json=None, url=""):
+        """Run query to Azure REST APIs."""
 
         try:
-            response = requests.post(
-                headers={
-                    "Authorization": "Bearer {}".format(self.access_token),
-                    "Content-Type": "application/json"
-                },
-                json={"query": query},
-                url="{}v1/apps/{}/query".format(
-                    self.api,
-                    self.application_id
+            # Define request headers
+            headers = {
+                "Authorization": "Bearer {}".format(self.access_token),
+                "Content-Type": "application/json"
+            }
+
+            # Do request
+            if method == "get":
```

Comment:
HTTP methods are case-sensitive so I think these should probably follow uppercase convention (GET/POST). OTOH, it works. :smile:

Reply:
Agreed. I changed this in commit 63983ebeba35fc29f8e1cf87874c968905ae52db. I used lowercase method names simply because requests-library uses lowercase function names for methods :)